src/sysupdate/sysupdate.c

```diff
@@ -744,6 +745,24 @@ static int context_make_online(Context **ret, const char *node) {
         return 0;
 }
 
+static int context_on_acquire_progress(const Transfer *t, const Instance *inst, unsigned percentage, void *userdata) {
+        const Context *c = ASSERT_PTR(userdata);
+        size_t i, n = c->n_transfers;
+        unsigned overall;
+
+        for (i = 0; i < n; i++)
+                if (c->transfers[i] == t)
+                        break;
+        assert(i < n); /* We should have found the index */
+
+        overall = (percentage / n) + (i * (100 / n));
+
+        log_debug("Transfer %zu/%zu is %u%% complete (%u%% overall).", i+1, n, percentage, overall);
+        return sd_notifyf(false, "X_IMPORT_PROGRESS=%u\n"
+                                 "STATUS=Updating to '%s' (%u%% complete).",
```

Comment:
maybe also report `X_IMPORT_FILES_DONE=` and `X_IMPORT_FILES_LEFT=` or so? and then report `i` and `n - i` on those? that's usefu I'd say and there's plenty space in the dgram